Sources/Workspace/Workspace.swift

```diff
@@ -2619,6 +2632,48 @@ extension Workspace {
             return path
         }
     }
+    
+    private func checkSourceControlIntegrity(package: PackageReference,
```

Comment:
would be interesting to see if we can move this into SourceControlPackageContainer to better encapsulate the logic, but its fine as is if too complicated

Reply:
https://github.com/apple/swift-package-manager/pull/3890/commits/454a52ac8e71cc0e708ca5aa878762c16f44342d